packages/web-components/fast-foundation/src/date-picker/date-picker.template.ts

```diff
@@ -0,0 +1,324 @@
+    return html`
+      <div class="picker" part="picker">
+        <div class="picker-title" part="picker-title">
+            <${button}
+                class="title-action ${x =>
+                    title.isInteractive ? "interactive-title" : ""}"
+                @click="${x => title.action()}"
+                @keydown="${(x, c) => title.action(c.event)}"
+            >
+                ${x => title.text}
+            </${button}>
+            <${button}
+                class="arrow"
+                part="arrow-previous"
+                @click="${x => previousAction()}"
+                @keydown="${(x, c) => previousAction(c.event)}"
+            >
+                &downarrow;
+            </${button}>
+            <${button}
+                class="arrow"
+                part="arrow-next"
+                @click="${x => nextAction()}"
+                @keydown="${(x, c) => nextAction(c.event)}"
+            >
+                &uparrow;
+            </${button}>
+        </div>
+      <${grid}
+        class="picker-grid"
+        part="picker-grid"
+        generate-header="none"
+    >
+        ${repeat(
+            x => items,
+            html`
+            <${row}
+                role="row"
+                role-type="default"
+                class="picker-row"
+                part="picker-row"
+                grid-template-columns="1fr 1fr 1fr 1fr"
+            >
+            ${repeat(
+                x => x,
+                html`
+                <${cell}
+                    tabindex="-1"
+                    class="picker-cell ${x => (x.selected ? "selected" : "")}"
+                    part="picker-cell"
+                    grid-column="${(x, c) => c.index + 1}"
+                    @click="${x => x.action()}"
+                    @keyup="${(x, c) => x.keyup(c.event)}"
+                >
+                ${x => x.text}
+              </${cell}>
+            `,
+                { positioning: true }
+            )}
+          </${row}>
+        `
+        )}
+      </${grid}>
+      ${reset}
+    </div>
+  `;
+};
+
+/**
+ *
+ * @param context - The date-picker class context
+ * @param definition - Date-picker options
+ * @returns - A ViewTemplate
+ * @public
+ */
+export const datePickerTemplate: (
+    Context: ElementDefinitionContext,
+    definition: DatePickerOptions
+) => ViewTemplate<DatePicker> = (
+    context: ElementDefinitionContext,
+    definition: DatePickerOptions
+) => {
+    const textField = context.tagFor(TextField);
+    const anchoredRegion = context.tagFor(AnchoredRegion);
+    const button = context.tagFor(Button);
+    const calendar = context.tagFor(Calendar);
+    const resetButton = html`<${button} class="reset-text" part="reset-text" @click="${x =>
+        x.resetCalendar()}">${x => x.resetText}</${button}>`;
+
+    return html`
+    <template
+        @mouseover="${x => (x.overFlyout = true)}"
+        @mouseout="${x => (x.overFlyout = false)}"
+    >
+        <${textField}
```

Comment:
Ah, I see we're wrapping a text field, which I like from an encapsulation perspective. I'm wondering about the slots though. I think we need a way to provide the `label` through the default slot and am also wondering about the inability to have `end` slot content that's _not_ the calendar icon.

Reply:
Added support for label as well as start and end slots. The calendar icon can also be swapped out through the date-picker options.